### PR TITLE
[mlir][vector] Extend mask calculation for vector.contract

### PR DESCRIPTION
**Pull request by banach-space**
Make sure that when calculating the expected mask for `vector.contract`,
scalable sizes are correctly taken into account.

Depends on: #65724

### CODE REVIEW

**Review by jsetoain — APPROVED** [review]
All looks good to me. It's all coming together very nicely! :-)

**Review by dcaballe — APPROVED** [review]

File: mlir/lib/Dialect/Vector/IR/VectorOps.cpp [hunk 1]
```diff
@@ -911,22 +912,27 @@ Type ContractionOp::getExpectedMaskType() {
 
   unsigned numVecDims = lhsIdxMap.getNumDims();
   SmallVector<int64_t> maskShape(numVecDims, ShapedType::kDynamic);
+  SmallVector<bool> maskShapeScalabledims(numVecDims, false);
```
Comment (c-rhodes) [inline mlir/lib/Dialect/Vector/IR/VectorOps.cpp:915]: `s/maskShapeScalabledims/maskShapeScalableDims`

Comment (banach-space) [reply mlir/lib/Dialect/Vector/IR/VectorOps.cpp]: Ta, see the latest fixup commit

File: mlir/test/Dialect/Vector/ops.mlir [hunk 2]
```diff
@@ -950,3 +979,22 @@ func.func @vector_scalable_extract(%sv: vector<[8]xi32>) {
...
+#matmat_accesses = [
+  affine_map<(i, j, k) -> (i, k)>,
+  affine_map<(i, j, k) -> (k, j)>,
+  affine_map<(i, j, k) -> (i, j)>
+]
+#matmat_trait = {
+  indexing_maps = #matmat_accesses,
+  iterator_types = ["parallel", "parallel", "reduction"]
+}
+func.func @matmul_masked_scalable(%arg0: vector<3x4xf32>,
```
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Dialect/Vector/ops.mlir]: Ta, see the latest fixup commit

File: mlir/test/Dialect/Vector/vector-contract-to-parallel-arith-transforms.mlir [hunk 3]
```diff
@@ -60,3 +60,4 @@ transform.sequence failures(propagate) {
     transform.apply_patterns.vector.lower_contraction lowering_strategy = "parallelarith"
   } : !transform.any_op
 }
+
```
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Dialect/Vector/vector-contract-to-parallel-arith-transforms.mlir]: Ta, see the latest fixup commit

**Review by c-rhodes — APPROVED** [review]

**Pull request merged**